src/test/java/hudson/security/LdapMultiEmbeddedTest.java

```diff
@@ -340,7 +340,7 @@ public void when_first_is_wrong_and_ignorable_and_login_on_second_then_login() t
                 allOf(containsString(FAILED_COMMUNICATION_WITH_LDAP_SERVER),
                         containsString(WILL_TRY_NEXT_CONFIGURATION),
                         containsString(planetExpress.getUrl())),
-                CoreMatchers.<Throwable>instanceOf(UserMayOrMayNotExistException2.class)));
+                CoreMatchers.<Throwable>instanceOf(AuthenticationServiceException.class)));
```

Comment:
Sorry but I do not remember if the specific exception type here is important. What is the full stack trace in the logs before and after the change? Maybe the only difference is that there is a new outer exception?

Reply:
`UserMayOrMayNotExistException2` seems important as it is used to say "I can not tell you that this users exists. or not"  and as such can allow the use of a token (wheras another excpetion would be treated as the form - this user does not exist).

At the very least I think some manual testing with user impersonation needs to occur when the server is down (before and after this change)